wooting-macro-backend/src/lib.rs

```diff
@@ -212,18 +217,29 @@ impl MacroData {
                             TriggerEventType::KeyPressEvent { data, .. } => {
                                 //TODO: optimize using references
                                 match data.len() {
-                                    0 => error!("A trigger key can't be zero...: {:#?}", data),
-                                    1 => output_hashmap
-                                        .entry(*data.first().unwrap())
-                                        .or_default()
-                                        .push(macros.clone()),
+                                    0 => {
+                                        return Err(Error::msg(format!("A trigger key can't be zero, aborting trigger generation: {:#?}", data).to_string()));
```

Comment:
This could be replaced with the `bail!` or `anyhow!` macros: https://docs.rs/anyhow/latest/anyhow/macro.bail.html

Reply:
Thanks, tried it here: [6f4b94a](https://github.com/WootingKb/wooting-macros/pull/194/commits/6f4b94a4c9bc70761f9724d73c2e240f2b562e98)